functions/imagemagick/index.js

```diff
@@ -30,115 +30,94 @@ const {BLURRED_BUCKET_NAME} = process.env;
-
-      // Blur the image using ImageMagick.
-      return new Promise((resolve, reject) => {
-        gm(tempLocalPath)
-          .blur(0, 16)
-          .write(tempLocalPath, (err, stdout) => {
-            if (err) {
-              console.error('Failed to blur image.', err);
-              reject(err);
-            } else {
-              resolve(stdout);
-            }
-          });
-      });
-    })
-    .then(() => {
-      console.log(`Image ${file.name} has been blurred.`);
-
-      // Upload result to a different bucket, to avoid re-triggering this function.
-      // You can also re-upload it to the same bucket + tell your Cloud Function to
-      // ignore files marked as blurred (e.g. those with a "blurred" prefix)
-      const blurredBucket = storage.bucket(blurredBucketName);
-
-      // Upload the Blurred image back into the bucket.
-      return blurredBucket
-        .upload(tempLocalPath, {destination: file.name})
-        .catch(err => {
-          console.error('Failed to upload blurred image.', err);
-          return Promise.reject(err);
-        });
-    })
-    .then(() => {
-      console.log(
-        `Blurred image has been uploaded to: gs://${blurredBucketName}/${file.name}`
-      );
-
-      // Delete the temporary file.
-      return new Promise((resolve, reject) => {
-        fs.unlink(tempLocalPath, err => {
+  try {
+    await file.download({destination: tempLocalPath});
+
+    console.log(`Image ${file.name} has been downloaded to ${tempLocalPath}.`);
+  } catch (err) {
+    console.error('Failed to download file.', err);
+    return Promise.reject(err);
+  }
+
+  try {
+    // Blur the image using ImageMagick.
+    await new Promise((resolve, reject) => {
+      gm(tempLocalPath)
+        .blur(0, 16)
+        .write(tempLocalPath, (err, stdout) => {
           if (err) {
-            reject(err);
+            console.error('Failed to blur image.', err);
+            return reject(err);
           } else {
-            resolve();
+            resolve(stdout);
           }
         });
-      });
     });
-}
+    console.log(`Image ${file.name} has been blurred.`);
+  } catch (err) {
+    console.error('Failed to blur image.', err);
+    return Promise.reject(err);
+  }
+
+  // Upload result to a different bucket, to avoid re-triggering this function.
+  // You can also re-upload it to the same bucket + tell your Cloud Function to
+  // ignore files marked as blurred (e.g. those with a "blurred" prefix)
+  const blurredBucket = storage.bucket(blurredBucketName);
+
+  // Upload the Blurred image back into the bucket.
+  try {
+    await blurredBucket.upload(tempLocalPath, {destination: file.name});
+    console.log(
+      `Blurred image has been uploaded to: gs://${blurredBucketName}/${file.name}`
+    );
+  } catch (err) {
+    console.error('Failed to upload blurred image.', err);
+    return Promise.reject(err);
+  }
+
+  // Delete the temporary file.
+  return new Promise((resolve, reject) => {
```

Comment:
I think it's cleaner practice to use [util.promisify()](https://nodejs.org/api/util.html#util_util_promisify_original)

Reply:
Turns out `promisify` doesn't work here. (No idea why - perhaps it doesn't have the right callback signature? 🤔)